src/officehours_api/tests.py

```diff
@@ -0,0 +1,132 @@
+        self.create_meeting([self.foo, self.bar, self.baz, self.hostoptout])
+        self.assertEqual(MockTwilio().messages.create.call_count, 2)
+        receivers = self.get_receivers(MockTwilio)
+        self.assertFalse(
+            receivers >=
+            {'+15555553333',}
+        )
+
+    @mock.patch('officehours_api.notifications.TwilioClient')
+    def test_first_meeting_doesnt_notify_attendees(self, MockTwilio: mock.MagicMock):
+        self.create_meeting([self.foo, self.bar, self.baz])
+        receivers = self.get_receivers(MockTwilio)
+        self.assertFalse(
+            receivers >=
+            {'+15555550000', '+15555550001', '+15555550002',}
+        )
+
+    @mock.patch('officehours_api.notifications.TwilioClient')
+    def test_second_meeting_doesnt_notify_hosts(self, MockTwilio: mock.MagicMock):
+        self.create_meeting([self.foo,])
+        MockTwilio.reset_mock()
+        self.create_meeting([self.bar, self.baz,])
+        receivers = self.get_receivers(MockTwilio)
+        self.assertFalse(
+            receivers >=
+            {'+15555551111', '+15555552222',}
+        )
+
+    @mock.patch('officehours_api.notifications.TwilioClient')
+    def test_first_meeting_removal_notifies_next_in_line(self, MockTwilio: mock.MagicMock):
+        m1 = self.create_meeting([self.foo,])
+        self.create_meeting([self.bar, self.baz])
+        MockTwilio.reset_mock()
+        m1.delete()
+        receivers = self.get_receivers(MockTwilio)
+        self.assertTrue(
+            receivers >=
+            {'+15555550001', '+15555550002',}
+        )
+
+    @mock.patch('officehours_api.notifications.TwilioClient')
+    def test_first_meeting_removal_doesnt_notify_optout(self, MockTwilio: mock.MagicMock):
+        m1 = self.create_meeting([self.foo,])
+        self.create_meeting([self.bar, self.baz, self.attendeeoptout,])
+        MockTwilio.reset_mock()
+        m1.delete()
+        receivers = self.get_receivers(MockTwilio)
+        self.assertFalse(
+            receivers >=
+            {'+15555550003',}
+        )
+
+    @mock.patch('officehours_api.notifications.TwilioClient')
+    def test_second_meeting_removal_notifies_none(self, MockTwilio: mock.MagicMock):
+        self.create_meeting([self.foo,])
+        m2 = self.create_meeting([self.bar, self.baz])
+        MockTwilio.reset_mock()
+        m2.delete()
+        receivers = self.get_receivers(MockTwilio)
+        self.assertFalse('+15555550001' in receivers)
+        self.assertFalse('+15555550002' in receivers)
+
+    @mock.patch('officehours_api.notifications.TwilioClient')
+    def test_first_meeting_removal_doesnt_notify_second_in_line(self, MockTwilio: mock.MagicMock):
+        m1 = self.create_meeting([self.foo,])
+        self.create_meeting([self.bar,])
+        self.create_meeting([self.baz,])
+        MockTwilio.reset_mock()
+        m1.delete()
+        receivers = self.get_receivers(MockTwilio)
+        self.assertFalse('+15555550002' in receivers)
```

Comment:
Would it make sense to add a test simulating how Twilio responds (and then testing how our application responds) if the phone number doesn't exist, or other exceptions? When my phone number was unverified, I was getting some weird behavior and Internal Server Errors that prevented me from loading a queue edit page.

Reply:
Yup, good thinking - I made some changes to how exceptions are handled (log the exception at error level with full stacktrace & continue) and was already looking into testing it.

Reply:
Added a couple tests to make sure it writes the exception to an ERROR logger and continues in those cases.